webapp-client/src/component/Navbar.js

```diff
@@ -2,6 +2,56 @@ import logo from '../lobster-icon.jpg';
 import React from 'react';
 
 export default class Navbar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loggedIn: false,
+      authInstance: null,
+    };
+    this.formatUser = this.formatUser.bind(this);
+    this.gapiSetState = this.gapiSetState.bind(this);
+    this.setLoggedIn = this.setLoggedIn.bind(this);
+    this.signOut = this.signOut.bind(this);
+  }
+
+  componentDidMount() {
+    window.gapi.load('auth2', this.gapiSetState);
+  }
+
+  gapiSetState() {
+    window.gapi.auth2.init({
+      client_id: process.env.REACT_APP_CLIENT_ID,
+    });
+    const authInstance = window.gapi.auth2.getAuthInstance();
+    this.setState({ authInstance });
+    authInstance.isSignedIn.listen(this.setLoggedIn);
+  }
+
+  setLoggedIn(loggedIn) {
+    this.setState({ loggedIn });
+  }
+
+  signOut() {
+    this.state.authInstance.signOut();
```

Comment:
Since authInstance is null by default, do you need to check if it is !null? Can window.gapi.auth2.init fail? Is [this](https://developers.google.com/identity/sign-in/web/reference#gapiauth2initparams) the one you are using? Because it looks like that returns a promise so it isn't going to be ready immediately. Also just curious where does window.gapi.auth2 get assigned?

Reply:
This is a really good catch! I added their then method, please see if it makes sense. window.gapi.auth2 gets assigned when a script called in the head of index.html assigns the global variable gapi. I have loaded that synchronously to try and make sure there are less issues